plugins/adr-backend/src/service/router.ts

```diff
@@ -133,5 +133,72 @@ export async function createRouter(
     }
   });
 
+  router.get('/image', async (req, res) => {
+    const urlToProcess = req.query.url as string;
+    if (!urlToProcess) {
+      res.statusCode = 400;
+      res.json({ message: 'No URL provided' });
+      return;
+    }
+
+    const imageTypeMap: Record<string, string> = {
+      png: 'image/png',
+      jpg: 'image/jpeg',
+      jpeg: 'image/jpeg',
+      gif: 'image/gif',
+      svg: 'image/svg+xml',
+      webp: 'image/webp',
+    };
+
+    const imageType = urlToProcess.match(/\.([a-z0-9]+)(\?.*)?$/i);
+    if (!(imageType && imageType[1])) {
+      res.statusCode = 400;
+      res.json({ message: 'No URL to image' });
+      return;
+    }
+    if (!imageTypeMap[imageType[1].toLowerCase()]) {
+      res.statusCode = 400;
+      res.json({ message: `Image type ${imageType[1]} is not supported` });
+      return;
+    }
+    const contentType = imageTypeMap[imageType[1].toLowerCase()];
+
+    const cachedFileContent = (await cacheClient.get(urlToProcess)) as {
+      data: string;
+      etag: string;
+    };
+
+    try {
+      const fileGetResponse = await reader.readUrl(urlToProcess, {
+        etag: cachedFileContent?.etag,
+      });
+
+      const fileBuffer = await fileGetResponse.buffer();
+      const data = fileBuffer.toString();
+
+      await cacheClient.set(urlToProcess, {
+        data,
+        etag: fileGetResponse.etag,
+      });
+
+      res.setHeader('Content-Type', contentType);
+      res.send(fileBuffer);
```

Comment:
There's a risk here that `contentType` is undefined, and that the `fileBuffer` is sent without an explicit content type, leading to an XSS vulnerability. We'd at least want a fallback here, but see the separate comment about making this a separate fetch in which case we might even make this base64 in a json response or something like it.